plugins/inputs/http_listener_v2/http_listener_v2.go

```diff
@@ -268,6 +269,16 @@ func (h *HTTPListenerV2) collectBody(res http.ResponseWriter, req *http.Request)
 		return nil, false
 	}
 
+	// Handle snappy request bodies
+	if req.Header.Get("Content-Encoding") == "snappy" {
+		bytes, err = snappy.Decode(nil, bytes)
+		if err != nil {
+			h.Log.Debug(err.Error())
+			badRequest(res)
+			return nil, false
+		}
+	}
+
```

Comment:
Is it possible to use the `snappy.NewReader` function and match the processing of `gzip`!?

Reply:
See comment below

Reply:
Could you maybe add a small comment here that we need the block-decode!?

Reply:
Yes, added

Reply:
Thanks!